### ISSUE TITLE
Installation issue on 3.12

### ISSUE THREAD

**Issue by cibere** [jcrist/msgspec#556]
### Description

So I'm trying to switch over to 3.12, and I'm having some issues with the installing msgspec.

Error:
```py
Collecting msgspec
  Using cached msgspec-0.18.2.tar.gz (208 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: msgspec
  Building wheel for msgspec (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for msgspec (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [32 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-cpython-312
      creating build\lib.win-amd64-cpython-312\msgspec
      copying msgspec\inspect.py -> build\lib.win-amd64-cpython-312\msgspec
      copying msgspec\json.py -> build\lib.win-amd64-cpython-312\msgspec
      copying msgspec\msgpack.py -> build\lib.win-amd64-cpython-312\msgspec
      copying msgspec\structs.py -> build\lib.win-amd64-cpython-312\msgspec
      copying msgspec\toml.py -> build\lib.win-amd64-cpython-312\msgspec
      copying msgspec\yaml.py -> build\lib.win-amd64-cpython-312\msgspec
      copying msgspec\_json_schema.py -> build\lib.win-amd64-cpython-312\msgspec
      copying msgspec\_utils.py -> build\lib.win-amd64-cpython-312\msgspec
      copying msgspec\_version.py -> build\lib.win-amd64-cpython-312\msgspec
      copying msgspec\__init__.py -> build\lib.win-amd64-cpython-312\msgspec
      copying msgspec\py.typed -> build\lib.win-amd64-cpython-312\msgspec
      copying msgspec\json.pyi -> build\lib.win-amd64-cpython-312\msgspec
      copying msgspec\msgpack.pyi -> build\lib.win-amd64-cpython-312\msgspec
      copying msgspec\structs.pyi -> build\lib.win-amd64-cpython-312\msgspec
      copying msgspec\__init__.pyi -> build\lib.win-amd64-cpython-312\msgspec
      UPDATING build\lib.win-amd64-cpython-312\msgspec/_version.py
      set build\lib.win-amd64-cpython-312\msgspec/_version.py to '0.18.2'
      running build_ext
      building 'msgspec._core' extension
      creating build\temp.win-amd64-cpython-312
      creating build\temp.win-amd64-cpython-312\Release
      creating build\temp.win-amd64-cpython-312\Release\msgspec
      "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\x86_amd64\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD -IC:\github\some-web-framework\venv\include -IC:\Users\[NAME]\AppData\Local\Programs\Python\Python312\include -IC:\Users\[NAME]\AppData\Local\Programs\Python\Python312\Include "-IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE" "-IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\ATLMFC\INCLUDE" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.10240.0\ucrt" /Tcmsgspec\_core.c /Fobuild\temp.win-amd64-cpython-312\Release\msgspec\_core.obj
      _core.c
      c:\users\[NAME]\appdata\local\programs\python\python312\include\pyconfig.h(230): fatal error C1083: Cannot open include file: 'basetsd.h': No such file or directory
      error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\BIN\\x86_amd64\\cl.exe' failed with exit code 2
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for msgspec
Failed to build msgspec
ERROR: Could not build wheels for msgspec, which is required to install pyproject.toml-based projects
```

**Comment by jcrist** [comment]
A quick google shows this error comes up in windows environments that aren't setup to properly compile CPython extension modules. There's a few stackoverflow posts that offer suggestions - perhaps one of them will be helpful?

Either way, I should have a pre-built wheel release for 3.12 out shortly, which will alleviate your issue.

**Comment by jcrist** [comment]
Version 0.18.3 has been released with wheels built for 3.12. Closing.

**Comment by gabrielgbs97** [comment]
Ihave aa similar issue:

```
cl.exe /c /nologo /O2 /W3 /GL /DNDEBUG /MD -IC:\Users\gbarcelo\source\repos\PythonREST\myenv\include "-IC:\Program Files\Python313\include" "-IC:\Program Files\Python313\Include" /Tcmsgspec\_core.c /Fobuild\temp.win32-cpython-313\Release\msgspec\_core.obj
      error: command 'cl.exe' failed: None
```

using MSVC C++ 2022.
